ic-agent/src/agent/http_transport/dynamic_routing/node.rs

```diff
@@ -0,0 +1,43 @@
+use url::Url;
+
+use crate::agent::ApiBoundaryNode;
+use anyhow::anyhow;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Node {
+    pub domain: String,
+}
+
+impl Node {
+    pub fn new(domain: &str) -> anyhow::Result<Self> {
+        if !is_valid_domain(domain) {
+            return Err(anyhow!("Invalid domain name {domain}"));
+        }
+        Ok(Self {
+            domain: domain.to_string(),
+        })
+    }
+}
+
+impl Node {
+    pub fn to_routing_url(&self) -> Url {
+        Url::parse(&format!("https://{}/api/v2/", self.domain)).expect("failed to parse URL")
+    }
+}
+
+impl From<&Node> for Url {
+    fn from(node: &Node) -> Self {
+        Url::parse(&format!("https://{}", node.domain)).expect("failed to parse URL")
+    }
+}
+
+impl From<&ApiBoundaryNode> for Node {
+    fn from(api_bn: &ApiBoundaryNode) -> Self {
+        Node::new(api_bn.domain.as_str()).unwrap()
+    }
+}
+
+pub fn is_valid_domain<S: AsRef<str>>(domain: S) -> bool {
+    // TODO
```

Comment:
I would just use the same as here: https://github.com/dfinity/ic/blob/master/rs/registry/canister/src/mutations/common.rs#L131-L155

Reply:
i think reusing the `Url` library some more concise:
```
pub fn is_valid_domain<S: AsRef<str>>(domain: S) -> bool {
    // Prepend scheme to make it a valid URL
    let url_string = format!("http://{}", domain.as_ref());
    Url::parse(&url_string).is_ok()
}
```